src/devices/Dhtxx/DhtBase.cs

```diff
@@ -269,7 +277,11 @@ internal virtual void ReadThroughI2c()
         /// <inheritdoc/>
         public void Dispose()
         {
-            _controller?.Dispose();
+            if (_shouldDispose)
+            {
```

Comment:
NIT: looks like here we are missing to also reassign _controller to Null, I know it wasn't introduced by your change but can we do that in case double Dispose is called?

Reply:
It's a readonly value, so I don't think it can be assigned to null?